manual/manual_circuit.rst

```diff
@@ -520,164 +655,253 @@ It is possible to specify small unitaries from ``numpy`` arrays and embed them d
 
-Another notable example that is common to many algorithms and high-level circuit descriptions is the exponential of a Pauli tensor: :math:`e^{-i \pi \theta P}` (:math:`P \in \{I, X, Y, Z\}^{\otimes n}`). These occur very naturally in Trotterising evolution operators and as common native device operations.
+Also in this category of synthesis boxes is :py:class:`DiagonalBox`. This allows synthesis of circuits for diagonal unitaries. 
+This box can be constructed by passing in a :math:`(1 \times 2^n)` numpy array representing the diagonal entries of the desired unitary matrix.
+
+Pauli Exponential Boxes and Phase Polynommials
+==============================================
+
+Another notable example that is common to many algorithms and high-level circuit descriptions is the exponential of a Pauli tensor: 
+
+.. math::
+    
+    \begin{equation}
+    e^{-i \frac{\pi}{2} \theta P}\,, \quad P \in \{I, X, Y, Z\}^{\otimes n}
+    \end{equation} 
+
+
+These occur very naturally in Trotterising evolution operators and sometimes as native device operations.
 
 .. jupyter-execute::
 
-    from pytket.circuit import Circuit, PauliExpBox
+    from pytket.circuit import PauliExpBox
     from pytket.pauli import Pauli
 
-    circ = Circuit(4)
-    circ.add_pauliexpbox(PauliExpBox([Pauli.X, Pauli.Y], 0.1), [0, 1])
-    circ.add_pauliexpbox(PauliExpBox([Pauli.Y, Pauli.X], -0.1), [0, 1])
-    circ.add_pauliexpbox(PauliExpBox([Pauli.X, Pauli.Y, Pauli.Y, Pauli.Y], 0.2), [0, 1, 2, 3])
-    circ.add_pauliexpbox(PauliExpBox([Pauli.Y, Pauli.X, Pauli.Y, Pauli.Y], -0.2), [0, 1, 2, 3])
+    # Construct PauliExpBox(es) with a list of Paulis followed by the phase
+    xyyz = PauliExpBox([Pauli.X, Pauli.Y, Pauli.Y, Pauli.Z], -0.2)
+    zzyx = PauliExpBox([Pauli.Z, Pauli.Z, Pauli.Y, Pauli.X], 0.7)
 
-In addition to the box types mentioned above ``pytket`` also supports a :py:class:`ToffoliBox` structure. This box type can be used to prepare an arbitrary permutation of the computational basis states using the :math:`\{\text{X},\text{CnX}\}` gateset.
+    pauli_circ = Circuit(5)
 
-In order to construct a :py:class:`ToffoliBox` the user must supply the desired permutation as a dictionary specifying the action of the box on different input basis states, where a key:value pair implies that the basis state key should be mapped to the basis state value.
+    pauli_circ.add_pauliexpbox(xyyz, [0, 1, 2, 3])
+    pauli_circ.add_pauliexpbox(zzyx, [1, 2, 3, 4])
 
-The given dictionary should provide permutations that correspond to complete cycles of basis states, i.e. if a basis state is present as a key then it must also be present as a value in the dictionary. If a valid permutation is supplied then a :py:class:`ToffoliBox` is constructed to perform the permutation and an error is thrown if the provided permutation is invalid.
+To understand what happens inside a :py:class:`PauliExpBox` lets take a look at the underlying circuit for :math:`e^{-i \frac{\pi}{2}\theta ZZYX}`
```

Comment:
```suggestion
To understand what happens inside a :py:class:`PauliExpBox` let's take a look at the underlying circuit for :math:`e^{-i \frac{\pi}{2}\theta ZZYX}`
```

Reply:
Yes, see the definition of `PauliExpBox` in https://cqcl.github.io/tket/pytket/api/optype.html#pytket.circuit.OpType .